File: Programmers/chaeeun/땅따먹기.py
def solution(land):

  for i in range(1, len(land)): # 행 인덱스
    for j in range(4): # 열 인덱스 - 모두 4열로 이루어져 있음
      land[i][j] += max(land[i-1][0:j] + land[i-1][j+1:])

  return max(land[-1])

File: Programmers/juyonLee/땅따먹기.cpp
//12913

#include <iostream>
#include <vector>
using namespace std;

int check(vector<int>& v, int idx)
{
    int num = 0;

    for(int i = 0; i < 4; i++)
    {
        if(i != idx)
            num = max(num, v[i]);
    }
    return num;
}

int solution(vector<vector<int> > land)
{
    int answer = 0;

    for(int i = 1; i < land.size(); i++)
    {
        for(int j = 0; j < 4; j++)
        {
            land[i][j] += check(land[i-1], j);
            
            answer = max(answer, land[i][j]);
        }
    }
    return answer;
}